src/lib/CheckType.hs

```diff
@@ -421,6 +418,19 @@ dictExprType e = case e of
   IxFin n -> do
     n' <- checkTypeE NatTy n
     liftM DictTy $ ixDictType $ TC $ Fin n'
+  ExplicitMethods ty methodFunNames args -> do
+    DictTy (DictType _ className params) <- checkTypeE TyKind (DictTy ty)
+    ClassDef _ _ pbs (SuperclassBinders Empty _) methodTys <- lookupClassDef className
+    forMZipped_ methodTys methodFunNames \(MethodType _ methodTy) methodFunName -> do
+      reqTy <- checkedApplyNaryAbs (Abs pbs methodTy) params
+      let args' = case reqTy of
+            Pi _ -> args
+            _ -> args ++ [UnitVal]
+      methodFunTy <- getType =<< substM methodFunName
+      actualTy  <- checkApp methodFunTy args'
+      checkAlphaEq reqTy actualTy
+      -- TODO: something special with thunks/nullary methods?
```

Comment:
You already do something special about them? Please clarify the TODO or remove it